src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,497 @@
+function.injective.module S to_prod_hom ext (λ x y, rfl)
+
+-- TODO: generalize to `algebra S 𝓜(𝕜, A)` once `continuous_linear_map.algebra` is generalized.
+instance : algebra 𝕜 𝓜(𝕜, A) :=
+{ to_fun := λ k,
+  { to_prod := algebra_map 𝕜 ((A →L[𝕜] A) × (A →L[𝕜] A)) k,
+    central := λ x y, by simp_rw [prod.algebra_map_apply, algebra.algebra_map_eq_smul_one,
+      smul_apply, one_apply, mul_smul_comm, smul_mul_assoc] },
+  map_one' := ext _ _ $ map_one $ algebra_map 𝕜 ((A →L[𝕜] A) × (A →L[𝕜] A)),
+  map_mul' := λ k₁ k₂, ext _ _ $ prod.ext (map_mul (algebra_map 𝕜 (A →L[𝕜] A)) _ _)
+    ((map_mul (algebra_map 𝕜 (A →L[𝕜] A)) _ _).trans (algebra.commutes _ _)),
+  map_zero' := ext _ _ $ map_zero $ algebra_map 𝕜 ((A →L[𝕜] A) × (A →L[𝕜] A)),
+  map_add' := λ _ _, ext _ _ $ map_add (algebra_map 𝕜 ((A →L[𝕜] A) × (A →L[𝕜] A))) _ _,
+  commutes' := λ _ _, ext _ _ $ prod.ext (algebra.commutes _ _) (algebra.commutes _ _).symm,
+  smul_def' := λ _ _, ext _ _ $ prod.ext (algebra.smul_def _ _)
+    ((algebra.smul_def _ _).trans $ algebra.commutes _ _) }
+
+@[simp] lemma algebra_map_to_prod (k : 𝕜) :
+  (algebra_map 𝕜 𝓜(𝕜, A) k).to_prod = algebra_map 𝕜 _ k := rfl
+lemma algebra_map_fst (k : 𝕜) : (algebra_map 𝕜 𝓜(𝕜, A) k).fst = algebra_map 𝕜 _ k := rfl
+lemma algebra_map_snd (k : 𝕜) : (algebra_map 𝕜 𝓜(𝕜, A) k).snd = algebra_map 𝕜 _ k := rfl
+
+/-!
+### Star structure
+-/
+
+section star
+
+variables [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A]
+
+/-- The star operation on `a : 𝓜(𝕜, A)` is given by
+`(star a).to_prod = (star ∘ a.snd ∘ star, star ∘ a.fst ∘ star)`. -/
+instance : has_star 𝓜(𝕜, A) :=
+{ star := λ a,
+  { fst := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.snd).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    snd := (((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A).comp a.fst).comp
+      ((starₗᵢ 𝕜 : A ≃ₗᵢ⋆[𝕜] A) : A →L⋆[𝕜] A),
+    central := λ x y, by simpa only [star_mul, star_star]
+      using (congr_arg star (a.central (star y) (star x))).symm } }
+
+@[simp] lemma star_fst (a : 𝓜(𝕜, A)) (b : A) : (star a).fst b = star (a.snd (star b)) := rfl
+@[simp] lemma star_snd (a : 𝓜(𝕜, A)) (b : A) : (star a).snd b = star (a.fst (star b)) := rfl
+
+instance : star_add_monoid 𝓜(𝕜, A) :=
+{ star_involutive := λ x, by {ext; simp only [star_fst, star_snd, star_star]},
+  star_add := λ x y, by {ext; simp only [star_fst, star_snd, add_fst, add_snd,
+    continuous_linear_map.add_apply, star_add]},
+  .. double_centralizer.has_star }
+
+instance : star_ring 𝓜(𝕜, A) :=
+{ star_mul := λ a b, by {ext; simp only [star_fst, star_snd, mul_fst, mul_snd, star_star,
+    continuous_linear_map.coe_mul, function.comp_app]},
+  .. double_centralizer.star_add_monoid }
+
+instance : star_module 𝕜 𝓜(𝕜, A) :=
+{ star_smul := λ k a, by {ext; exact star_smul _ _},
+  .. double_centralizer.star_add_monoid }
+
+end star
+
+/-!
+### Coercion from an algebra into its multiplier algebra
+-/
+
+noncomputable instance : has_coe_t A 𝓜(𝕜, A) :=
```

Comment:
Does this form a non-equal diamond with the `algebra_map` coercion if `A = 𝕜`? If no then it would be nice to prove that they're equal (definitionally or otherwise), and if not it might be worth a warning comment.

Reply:
They are propositionally but not definitionally equal and I have added a warning comment. In addition, I proved a lemma in order to convert between them if necessary.